packages/sdk/src/types/runtime/dialog.ts

```diff
@@ -0,0 +1,86 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { EventEmitter } from 'events';
+
+import { Context, GroupMask, Texture, User } from '.';
+
+/** The types of dialog boxes */
+export enum DialogType {
+	/** A dialog with text, an icon, and an OK button */
+	Message = 0,
+	/** A dialog with text, an icon, and OK and Cancel buttons */
+	Confirm,
+	/** A dialog with prompt text, a text input field, and OK and Cancel buttons */
+	Prompt
+}
+
+
+export enum DialogIcon {
+	/** A speech bubble */
+	Info = 0,
+	/** A yellow exclamation point */
+	Warning,
+	/** A red exclamation point */
+	Error,
+	/** A question mark */
+	Question
+}
+
+/** Pop a message to users, and get a response */
+export class Dialog extends EventEmitter {
+	/**
+	 * Create a new dialog box to present to users.
+	 * @param context An active MRE context.
+	 * @param options.type How to present the choice to users.
+	 * @param options.text The message to present.
+	 * @param options.recipients Who to send the message to. Defaults to all users.
+	 * @param options.icon A graphic to display next to the text. Defaults to none.
+	 * @param options.persistent Send this message to late-joining users. Defaults to false.
+	 */
+	public constructor(private context: Context, options: {
+		type: DialogType,
+		text: string,
+		recipients?: GroupMask | User[] | User,
+		icon?: DialogIcon | Texture,
+		persistent?: boolean
+	}) {
+		super();
+	}
+
+	/** Convenience method for broadcasting a message */
+	public static Message(context: Context, text: string): Dialog { }
```

Comment:
I think the fluent API mentioned above could achieve the same goal as these convenience methods, but with more flexibility. Your thoughts?

Reply:
Versus `(new Dialog(context, options)).show()`? Maybe, but it might confuse people that they're interchangeable.

---
In reply to: [320858387](https://github.com/microsoft/mixed-reality-extension-sdk/pull/444#discussion_r320858387) [](ancestors = 320858387)